mlflow/protos/webhooks.proto

```diff
@@ -0,0 +1,258 @@
+syntax = "proto2";
+
+package mlflow;
+
+import "scalapb/scalapb.proto";
+import "databricks.proto";
+
+option java_package = "org.mlflow.api.proto";
+option java_generate_equals_and_hash = true;
+option py_generic_services = true;
+option (scalapb.options) = {
+  flat_package: true,
+};
+
+// Webhook service for Model Registry
+service WebhookService {
+  rpc createWebhook (CreateWebhook) returns (CreateWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "POST",
+        path: "/mlflow/webhooks"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Create Webhook",
+    };
+  }
+
+  rpc listWebhooks (ListWebhooks) returns (ListWebhooks.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/webhooks"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "List Webhooks",
+    };
+  }
+
+  rpc getWebhook (GetWebhook) returns (GetWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "GET",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Get Webhook",
+    };
+  }
+
+  rpc updateWebhook (UpdateWebhook) returns (UpdateWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "PATCH",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Update Webhook",
+    };
+  }
+
+  rpc deleteWebhook (DeleteWebhook) returns (DeleteWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "DELETE",
+        path: "/mlflow/webhooks/{webhook_id}"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Delete Webhook",
+    };
+  }
+
+  rpc testWebhook (TestWebhook) returns (TestWebhook.Response) {
+    option (rpc) = {
+      endpoints: [{
+        method: "POST",
+        path: "/mlflow/webhooks/{webhook_id}/test"
+        since { major: 2, minor: 0 },
+      }],
+      visibility: PUBLIC,
+      rpc_doc_title: "Test Webhook",
+    };
+  }
+}
+
+// Webhook status enumeration
+enum WebhookStatus {
+  ACTIVE = 1;     // Webhook is active and receiving events
+  DISABLED = 2;   // Webhook is disabled and not receiving events
+  DELETED = 3;    // Webhook has been soft-deleted and is no longer active
```

Comment:
Will we support recovering webhook from DELETED to ACTIVE?

Reply:
We're unlikely to support it.

Reply:
If so, do we need this DELETED status? Seems useless